app/routes/circulars.moderation._index.tsx

```diff
@@ -31,45 +39,100 @@ export async function loader({ request }: LoaderFunctionArgs) {
   }
 }
 
+export async function action({ request }: ActionFunctionArgs) {
+  const user = await getUser(request)
+  if (!user || !user.groups.includes(moderatorGroup))
+    throw new Response(null, { status: 403 })
+  const data = await request.formData()
+  const selectedItems: CircularChangeRequestKeys[] = data
+    .getAll('bulkItems')
+    .map((x) => {
+      const splitValue = x.toString().split('::')
+      return {
+        circularId: parseFloat(splitValue[0]),
+        requestorSub: splitValue[1],
+      }
+    })
+
+  await bulkDeleteChangeRequests(selectedItems, user)
+  return null
+}
+
 export default function () {
   const { changeRequests } = useLoaderData<typeof loader>()
+  const [disableSubmit, setDisableSubmit] = useState(true)
+  const [selectedList, setSelectedList] = useState<string[]>([])
+
+  function checkboxOnChange(id: string, checked: boolean) {
+    if (checked) {
+      setSelectedList([...selectedList, id])
+    } else {
+      setSelectedList(selectedList.filter((x) => x !== id))
+    }
+  }
+
+  function setBool() {
+    setDisableSubmit(selectedList.length === 0)
+  }
```

Comment:
You can probably achieve a simpler state update if you just store the _number_ of selected checkboxes: it's initially zero; you increment it whenever a checkbox goes from unchecked to checked; you decrement it whenever a checkbox goes from checked to unchecked.

Reply:
Good call, that worked much better and cleaned up the code a bit too!